x/profiles/keeper/alias_functions.go

```diff
@@ -211,6 +210,32 @@ func (k Keeper) GetApplicationLinks(ctx sdk.Context) []types.ApplicationLink {
 	return links
 }
 
+// IterateExpiringApplicationLinks iterates through all the expiring application links references.
+// The key will be skipped and deleted if the application link has already been deleted.
+func (k Keeper) IterateExpiringApplicationLinks(ctx sdk.Context, fn func(index int64, link types.ApplicationLink) (stop bool)) {
+	store := ctx.KVStore(k.storeKey)
+
+	iterator := sdk.KVStorePrefixIterator(store, types.ApplicationLinkExpiringTimePrefix(ctx.BlockTime()))
```

Comment:
It is needed the block produced at the exactly time when applinks expired, right? 
If yes, it will not work properly because the block time is not always at time, says: a applink expired at `time.Date(2022, 1, 1, 00, 00, 00, 000, time.UTC)`, but actually the block might be produced at `time.Date(2022, 1, 1, 00, 00, 00, 001, time.UTC)`.

Reply:
I think iterating the `ExpiringAppLinkTimePrefix` in the asc way then deleting the applinks until the time is larger the given time would solve the issue since key in the db is sorted.